tools/autograd/templates/Functions.cpp

```diff
@@ -852,6 +852,23 @@ Tensor cholesky_backward(Tensor grad, bool upper, Tensor L) {
   return grad_input.add(grad_input.transpose(-1, -2)).mul_(0.5);  // Symmetrizing the gradient
 }
 
+Tensor cholesky_mod_backward(Tensor grad, bool upper, Tensor L, Tensor jitter) {
+  // L_tag = L - I * jiter[0], then use cholesky_backward
+  Tensor L_tag = L;
+  if (jitter.defined()) {
+    L_tag.copy_(L);
+    if (jitter.dim() > 0) {
+        std::cout << "jitter dim " << jitter.dim() << ", size[0] " << jitter.size(0) << std::endl;
+        L_tag.diagonal().add_(jitter[0]);
+    }
+    else {
+        L_tag.diagonal().add_(jitter);
+    }
+
+  }
+  return cholesky_backward(grad, upper, L_tag);
+}
```

Comment:
Hmm, why `L_tag = L - I * jiter[0]`?

Reply:
If we assume that `jitter` is a constant, then `A + e* I = L L^T` implies that `dA = dL L^T + L dL^T`, so the backward formula should be the same as for classic Cholesky.

Reply:
makes sense. I was making this too hard for no reason.